### PR TITLE
Add Continuous Integration with Travis

### LINKED ISSUE #2: Implement Continuous Integration pipeline
We should add Travis CI support. In this case though it is not a trivial task since there are a lot of dependencies from sources (mainly the ignition libraries).

On the long run we can use their binary packages, but at the current stage the best solution would exploit a docker image.

### PR DESCRIPTION
**Pull request by diegoferigo**
Closes #2 
Depends on #22 

This PR adds initial Travis CI support. During my tests I noticed the following strange behaviors that will be addressed in future PRs:

- macOS fails to find shared libraries dynamically linked to the swig library. I temporarily allowed failures on macOS builds.
- Pytest fails when clang is used on Linux. I temporarily disabled clang builds.

### CODE REVIEW

File: .ci/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    make -j8 &&\
+    make install &&\
+    rm -r /tmp/swig
+
+# Install ignition robotics
+RUN echo "deb http://packages.osrfoundation.org/gazebo/ubuntu-stable `lsb_release -cs` main" \
+        > /etc/apt/sources.list.d/gazebo-stable.list &&\
+    echo "deb http://packages.osrfoundation.org/gazebo/ubuntu-prerelease `lsb_release -cs` main" \
+        > /etc/apt/sources.list.d/gazebo-prerelease.list &&\
+    echo "deb http://packages.osrfoundation.org/gazebo/ubuntu-nightly `lsb_release -cs` main" \
```
Comment (traversaro) [inline .ci/Dockerfile]: I thought just  adding `ubuntu-nightly` should  be enough.

Comment (diegoferigo) [reply .ci/Dockerfile]: I thought it was enough too, but the repo is not self contained and nightly packages depend on other that are contained only in the other two. Not sure if this was intentional or just a packaging mistake. For the time being I'm fine to keep them all enabled, I wish to switch as soon as possible to using only the stable channel.

**Comment by diegoferigo** [comment]
@traversaro Do you have any ideas on why we have [this](https://travis-ci.org/robotology/gym-ignition/jobs/530809794#L2488-L2499) error in the swig bindings on macOS? I tried many combinations but I didn't manage to address this problem.

More that enabling RPATH and using `DYLD_LIBRARY_PATH` I am clueless.

**Comment by diegoferigo** [comment]
It seems to be a RPATH-related problem. The `_gympp.so` libraries has this on osx:

```
/Users/travis/local/lib/gympp/bindings/_gympp.so:
	@rpath/libGymFactory.dylib (compatibility version 0.0.0, current version 0.0.0)
	@rpath/libIgnitionEnvironment.dylib (compatibility version 0.0.0, current version 0.0.0)
	@rpath/libRobotSingleton.dylib (compatibility version 0.0.0, current version 0.0.0)
	@rpath/libGazeboWrapper.dylib (compatibility version 0.0.0, current version 0.0.0)
	/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/Python (compatibility version 3.7.0, current version 3.7.0)
	@rpath/libgympp.dylib (compatibility version 0.0.0, current version 0.0.0)
	/usr/local/opt/ignition-common3/lib/libignition-common3.3.dylib (compatibility version 3.0.0, current version 3.1.0)
	/usr/local/opt/ossp-uuid/lib/libuuid.16.dylib (compatibility version 17.0.0, current version 17.22.0)
	@rpath/libignition-gazebo2.2.dylib (compatibility version 2.0.0, current version 2.0.0)
	/usr/local/opt/ignition-fuel-tools3/lib/libignition-fuel_tools3.3.dylib (compatibility version 3.0.0, current version 3.1.0)
	...
```

I managed to get most libraries to find each other by properly adding the lib dirs in CMake:

https://github.com/robotology/gym-ignition/blob/ae6168c98de7a961fb1329f8e59337fce5cf30aa/CMakeLists.txt#L67-L75

Now the problem transferred to something related to the ignition libraries:

```
test_bindings.py:12: in <module>
    import gym_ignition
/usr/local/lib/python3.7/site-packages/gym_ignition/__init__.py:11: in <module>
    import gympp
/Users/travis/local/lib/gympp/bindings/gympp.py:15: in <module>
    import _gympp
E   ImportError: dlopen(/Users/travis/local/lib/gympp/bindings/_gympp.so, 10): Symbol not found: _rmtMetal_usGetTime
E     Referenced from: /usr/local/opt/ignition-common3/lib/libignition-common3-profiler.3.dylib
E     Expected in: flat namespace
E    in /usr/local/opt/ignition-common3/lib/libignition-common3-profiler.3.dylib
```

**Comment by traversaro** [comment]
I am a bit confused. Shouldn't the test be using the **build** version of the used libraries, as long  as they are built by `gym-ignition` as well? Having said that, if you want to test this on a real machine, you can use the iMac that we have in the lab.

**Comment by diegoferigo** [comment]
> I am a bit confused. Shouldn't the test be using the build version of the used libraries, as long as they are built by gym-ignition as well?

This is the classic test of C++ code, I agree. In this case though, only the python bindings are tested. Since the installation process in python is not that straightforward (at least to me, coming from C++), I thought to test the installed version of the python package, and by consequence, the installed version of the libraries.

It is surely possible to test the installed python package and set the system in such a way that the build libraries are used. Do you see any particular advantage of doing it in this way instead of testing the installed files? What I do not want to do is using the in-source python files since the python include structure is quite tricky, and using the in-source files might prevent to catch related problems.

**Comment by traversaro** [comment]
> It is surely possible to test the installed python package and set the system in such a way that the build libraries are used. Do you see any particular advantage of doing it in this way instead of testing the installed files? What I do not want to do is using the in-source python files since the python include structure is quite tricky, and using the in-source files might prevent to catch related problems.

Ack, as long as it is correctly documented for new developers  that is ok.

**Comment by diegoferigo** [comment]
Failure in osx is the same also from the build folder. It might be something related to the ignition libraries, not sure what this `_rmtMetal_usGetTime` symbol is in the `libignition-common3-profiler.3.dylib` shared library.

If there are no objections, I merge this initial Travis CI support as it is.

**Pull request merged**